### PR TITLE
tests: add go.mod file to isolate

### PR DESCRIPTION
**Pull request by jmank88**
This PR proposes isolating `tests/` as a separate module, like we did in the core repo. This removes more than 100 dependencies from the main `go.mod`!
[Solana version](https://github.com/smartcontractkit/chainlink-solana/pull/334)

Does CI need updating?

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,19 +4,12 @@ go 1.18
 
 require (
 	github.com/cosmos/cosmos-sdk v0.44.5
-	github.com/ethereum/go-ethereum v1.10.16
```
Comment (jmank88) [inline go.mod:7]: 🔥

**Review by aalu1418 — DISMISSED** [review]
🔥

**Review by krebernisak — CHANGES_REQUESTED** [review]
Please add a README to the `./tests` folder explaining the split.

**Comment by jmank88** [comment]
Added readme and created #283 to follow up.

**Review by aalu1418 — APPROVED** [review]
🔥

**Review by krebernisak — APPROVED** [review]
👏

**Pull request merged**